lib/li_box4d.c

```diff
@@ -0,0 +1,45 @@
+/***********************************************************************
+* li_box4d.c
+*
+***********************************************************************/
+
+#include "pc_api_internal.h"
+#include "li_api_internal.h"
+
+/**
+* Apply an affine transformation to a box4d.
+*/
+LIBOX4D *
+li_box4d_affine(
+	const LIBOX4D *ibox,
+	double a, double b, double c,
+	double d, double e, double f,
+	double g, double h, double i,
+	double xoff, double yoff, double zoff)
+{
+	LIMAT43 amat;
+	LIVEC3 vec, rvec;
+	LIBOX4D *obox = pcalloc(sizeof(LIBOX4D));
+
+	li_matrix_43_set(amat, a, b, c, xoff, d, e, f, yoff, g, h, i, zoff);
+
+	vec[0] = ibox->xmin;
+	vec[1] = ibox->ymin;
+	vec[2] = ibox->zmin;
```

Comment:
Having a box defined with a float[4][2] element rather than 8 float elements would allow you to get 2 corners as a LIVEC3 with a simple cast and get all coners using index arithmetics (eg in a for loop).